kube-runtime/src/controller/mod.rs

```diff
@@ -391,7 +390,29 @@ where
         I::IntoIter: Send,
     {
         let other_watcher = trigger_with(try_flatten_touched(watcher(api, lp)), mapper);
-        self.selector.push(other_watcher.boxed());
+        self.trigger_selector.push(other_watcher.boxed());
+        self
+    }
+
+    /// Trigger a reconciliation for all managed objects whenever `trigger` emits a value
+    ///
+    /// For example, this can be used to reconcile all objects whenever the controller's configuration changes.
+    pub fn reconcile_all_on(mut self, trigger: impl Stream<Item = ()> + Send + Sync + 'static) -> Self {
```

Comment:
Oh, this looks surprisingly simple!

Any chance you can put an example here for it? (can mark it `no_run` if it pulls in too much)

Reply:
74209651b4f42be3f798cb4f5ce4844f81ff8703

Never actually runs (the `Controller` is never polled) since that would require a K8s cluster to run against.

Reply:
But it illustrates the use case well! Thank you.

Now I guess users can wrap the stream in whatever read-write protection they want and push an item into it from a http endpoint / whatever hook they want.